slither/detectors/functions/codex.py

```diff
@@ -52,29 +53,63 @@ def _detect(self) -> List[Output]:
         openai.api_key = api_key
 
         for contract in self.compilation_unit.contracts:
-            prompt = "Is there a vulnerability in this solidity contracts?\n"
```

Comment:
Above in line 40, the `import openai` inside of this function was problematic when I tried to extract the OpenAI query logic into a separate function.  I think it scopes the import to the function so I wasn't able to access it.  

Maybe we can move the `import` (wrapped in a try/except) to the top level?  We could still do something here that checks whether it's been installed.